lib/query.js

```diff
@@ -3810,6 +3810,7 @@ function _completeManyLean(schema, docs, path, opts) {
 
 Query.prototype._mergeUpdate = function(doc) {
   if (doc == null || (typeof doc === 'object' && Object.keys(doc).length === 0)) {
+    this._update = {};
```

Comment:
I don't think we need this. Another solution would be to take
```
 if (!this._update) {
    this._update = Array.isArray(doc) ? [] : {};
  }
```
which is just below this `if` check and move it to the top of the `mergeUpdate` function. Tried this locally and it seems to fix the problem, we also wouldn't need to add a new line of code to this. 

This solution is safe since `Array.isArray(arg)` has null checks, so `doc == null` wouldn't be an issue.

Reply:
Actually thought about this, but for some reason I saw this snippet was inside the other if statement (bad attention span and was at the end of the day)

Reply:
I don't see a resolve button so I think you'll have to resolve this comment, all looks good now 👍🏼